src/session_impl.cpp

```diff
@@ -1151,27 +1151,43 @@ namespace {
 	void session_impl::queue_tracker_request(tracker_request& req
 		, std::weak_ptr<request_callback> c)
 	{
-		req.listen_port = listen_port();
+#if TORRENT_USE_I2P
+		if (!m_settings.get_str(settings_pack::i2p_hostname).empty())
+		{
+			req.i2pconn = &m_i2p_conn;
+		}
+#endif
+
 		if (m_key) req.key = m_key;
 
 #ifdef TORRENT_USE_OPENSSL
-		// SSL torrents use the SSL listen port
-		// TODO: 2 this need to be more thought through. There isn't necessarily
-		// just _one_ SSL listen port, which one we use depends on which interface
-		// we announce from.
-		if (req.ssl_ctx) req.listen_port = ssl_listen_port();
+		bool use_ssl = req.ssl_ctx != nullptr;
 		req.ssl_ctx = &m_ssl_ctx;
 #endif
 
-#if TORRENT_USE_I2P
-		if (!m_settings.get_str(settings_pack::i2p_hostname).empty())
+		if (req.outgoing_socket)
```

Comment:
I can't find any place in this patch where this member is assigned (other than in the else-block below)